2016-04/april-7.md

```diff
@@ -0,0 +1,129 @@
+* We need to figure out how to where to put “unreleased” entries, how to handle merge conflicts, etc.
+* Leaning towards having an oncall or whoever merges things updating the changelog.
+* What if whenever we sync React master internally to Facebook (once a couple of weeks), we also generate the changelog?
+  * This means that whoever does the sync has to do a little more work.
+  * Combination of the batched approach + human discretion.
+* Will start this process with the next internal sync.
+
+### Browser Tests
+
+* React attempts to normalize behavior across browsers but we don’t have any browser tests. ([#5703](https://github.com/facebook/react/issues/5703))
+* We used to have them with Sauce Labs but they were super flaky and we removed them. ([#4393](https://github.com/facebook/react/pull/4393))
+* Most of our tests aren’t tied to Jest and we should be able to get them running in the browser again.
+* Should we run the existing tests, or are there new types of things we want to test?
+  * The types of things we care about with browser tests are often different.
+  * Usually we don’t need Jest-isms like clearing the module registry in these tests as they only work with public API.
+  * Example: Input selection should not jump to end of input when focusing a controlled input (or something like this).
+* Should we be worried tests will get flaky again?
+  * We’ll start with a few very focused tests and see how it goes.
+* Let’s decide on a test runner and write some tests.
+  * Let’s pick a recent regression and try to write a test for it.
+  * [Jim](https://github.com/jimfb) will work on this.
+
+
+### Attributes vs Properties
+
+* In React 15, we switched to using DOM attributes by default and using properties only in special cases.
+* This fixed a few issues but then introduced a few other issues.
+* While we patched over those regressions and 15 looks good, there is no consensus yet on whether this was the right call.
+
+#### Reasons to Use Attributes over Properties
+
+* Attributes currently match our desired behavior for more or less all supported props, including SVG. In the future properties might support more structured data but there’s nothing there we _currently_ want to take advantage of.
+* 0.14 already created HTML strings on initial render, and attributes are the closest alternative.
+* If we use attributes for everything we could just drop the whitelist completely (and treat only form components specially), which we may or may not want to do but it seems appealing.
+* For the specific regression that relying on attributes introduced ([not setting `value=""` on option tags](https://github.com/facebook/react/issues/6219)), we _can’t_ use properties because there is a semantic difference between `value=""` and no attribute, but `.value` is `''` in both cases.
+* We’re not currently invested into designing a rich API for React DOM props right now so less maintenance burden there seems beneficial.
```

Comment:
(It's _because_ we're not actively doing this that less maintenance is appealing.)